pkg/config/config_factory_test.go

```diff
@@ -49,6 +53,7 @@ func TestGetCfgFactoryConfig(t *testing.T) {
 		provider := GetConfigProvider()
 		_, err = provider.Get(context.Background(), factories)
 		require.Error(t, err)
+		resetFlag()
```

Comment:
I would suggest doing this before the `require.Error()` call, possibly using [`t.Cleanup()`](https://pkg.go.dev/testing#T.Cleanup).  I believe that on failure `require.*` will prevent continuation of the test and thus this cleanup won't happen.